ResearchKit/Common/ORKAxisView.m

```diff
@@ -0,0 +1,135 @@
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+
+ 
+#import "ORKAxisView.h"
+
+
+@interface ORKAxisView ()
+
+@property (nonatomic, strong) NSMutableArray *titleLabels;
+
+@property (nonatomic) ORKGraphAxisType axisType;
+
+@end
+
+
+@implementation ORKAxisView
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        [self sharedInit];
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    self = [super initWithFrame:frame];
+    if (self) {
+        [self sharedInit];
+    }
+    return self;
+}
+
+- (void)sharedInit {
+    _titleLabels = [NSMutableArray new];
+    _leftOffset = 0;
+}
+
+- (void)layoutSubviews {
+    CGFloat segmentWidth = CGRectGetWidth(self.bounds) / (self.titleLabels.count - 1);
+    CGFloat labelWidth = segmentWidth;
+    
+    CGFloat labelHeight = (self.axisType == ORKGraphAxisTypeX) ? CGRectGetHeight(self.bounds) * 0.77 : 20;
+    
+    for (NSUInteger i = 0; i < self.titleLabels.count; i++) {
+        
+        CGFloat positionX = (self.axisType == ORKGraphAxisTypeX) ? (self.leftOffset + i * segmentWidth) : 0;
+        
+        if (i == 0) {
+            // Shift the first label to accomodate the month text.
+            positionX -= self.leftOffset;
+        }
+        
+        UILabel *label = (UILabel *)self.titleLabels[i];
+        
+        if (label.text) {
+            labelWidth = [label.text boundingRectWithSize:CGSizeMake(CGFLOAT_MAX, labelHeight)
+                                                  options:(NSStringDrawingUsesFontLeading | NSStringDrawingUsesLineFragmentOrigin)
+                                               attributes:@{NSFontAttributeName:label.font}
+                                                  context:nil].size.width;
+            labelWidth = MAX(labelWidth, 15);
+            labelWidth += self.landscapeMode ? 14 : 8; //padding
+        }
+        
+        if (i == 0) {
+            label.frame  = CGRectMake(positionX, (CGRectGetHeight(self.bounds) - labelHeight) / 2, labelWidth, labelHeight);
+        } else {
+            label.frame  = CGRectMake(positionX - labelWidth / 2, (CGRectGetHeight(self.bounds) - labelHeight) / 2, labelWidth, labelHeight);
+        }
+        
+        if (i == self.titleLabels.count - 1) {
+            // Last label.
+            label.textColor = [UIColor whiteColor];
+            label.backgroundColor = self.tintColor;
+            label.layer.cornerRadius = CGRectGetHeight(label.frame) / 2;
+            label.layer.masksToBounds = YES;
+        }
+    }
+}
+
+- (void)setupLabels:(NSArray *)titles forAxisType:(ORKGraphAxisType)type {
+    self.axisType = type;
+    
+    for (NSUInteger i = 0; i < titles.count; i++) {
+        
+        UILabel *label = [UILabel new];
+        label.text = titles[i];
+        label.font = self.isLandscapeMode ? [UIFont fontWithName:@"Helvetica-Light" size:19.0] : [UIFont fontWithName:@"Helvetica-Light" size:12.0];
```

Comment:
Let's leave this font non-configurable for now, but use `systemFontOfSize:12.0`. That way we'll get the new *San Francisco* font on *iOS 9*.